kolibri/core/assets/src/api_resource.js

```diff
@@ -39,24 +42,87 @@ class Model {
    * returns, otherwise reject is called with the response object.
    */
   fetch(params = {}, force = false) {
-    if (!force && this.synced) {
-      return Promise.resolve(this.attributes);
-    }
-    this.synced = false;
-    return new Promise((resolve, reject) => {
-      // Do a fetch on the URL.
-      client({ path: this.url, params }).then((response) => {
-        // Set the retrieved Object onto the Model instance.
-        this.set(response.entity);
-        // Flag that the Model has been fetched.
-        this.synced = true;
-        // Resolve the promise with the attributes of the Model.
-        resolve(this.attributes);
-      }, (response) => {
-        logging.error('An error occurred', response);
-        reject(response);
+    const promise = new Promise((resolve, reject) => {
+      Promise.all(this.promises).then(() => {
+        if (!force && this.synced) {
+          resolve(this.attributes);
+        } else {
+          this.synced = false;
+          // Do a fetch on the URL.
+          client({ path: this.url, params }).then((response) => {
+            // Set the retrieved Object onto the Model instance.
+            this.set(response.entity);
+            // Flag that the Model has been fetched.
+            this.synced = true;
+            // Resolve the promise with the attributes of the Model.
+            resolve(this.attributes);
+            // Clean up the reference to this promise
+            this.promises.splice(this.promises.indexOf(promise), 1);
+          }, (response) => {
+            logging.error('An error occurred', response);
```

Comment:
Hmm, my understanding is that if one fails, the rest will cascade fail (and I think that was our desired behaviour, too, as the later ones may have been contingent).

    The Promise.all(iterable) method returns a promise that resolves when all of the promises in the
    iterable argument have resolved, or rejects with the reason of the first passed promise that rejects.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all